### PR TITLE
DDCE-656: Changes to governing document details

### PR DESCRIPTION
**Pull request by abhijeetardale**
# DDCE-656

##Bug fix / New feature (delete as appropriate)

Please include a summary / description of the change and which issue it fixes.  Include any relevant user needs, context or links to other PRs related to this PR (eg. acceptance tests, environment config).

## PR Suggestions
- Have you looked at the JIRA story to make sure all Acceptance Criteria has been met?
- Have you done a visual check of the changes?
- Is there any still commented or unused code? Lingering printlns?
- Have things been implemented in a complicated way?
- Are the test still over the coverage threshold?
- Does the compiler throw a lot of warnings? 
- Have methods and tests been named clearly?
- Were there any changes in the config? Do changes need to be made in app-config-???

## Checklist PR Raiser
 - [x]  I've made every effort to commit high quality, clean code and I have executed relevant static analyses to be sure
 - [x]  I've included appropriate tests with any code I've added (Unit, Integration, Acceptance etc.)
 - [x]  I've executed the acceptance test pack locally to ensure there are no functional regressions (Performance tests where relevant)
 - [x]  I've squashed my commits - including the JIRA issue number in the commit message
 - [ ]  I've run a dependency check to ensure all dependencies are up to date

## Checklist PR Reviewer
 - [ ]  I've made every effort to commit high quality, clean code and I have executed relevant static analyses to be sure
 - [ ]  I've included appropriate tests with any code I've added (Unit, Integration, Acceptance etc.)
 - [ ]  I've executed the acceptance test pack locally to ensure there are no functional regressions (Performance tests where relevant)
 - [ ]  I've squashed my commits - including the JIRA issue number in the commit message
 - [ ]  I've run a dependency check to ensure all dependencies are up to date

### CODE REVIEW

**Comment by abhijeetardale** [comment]
Linked PR please
https://github.com/hmrc/charities-registration-journey-tests/pull/128

File: app/navigation/DocumentsNavigator.scala [hunk 1]
```diff
@@ -85,26 +89,34 @@ class DocumentsNavigator @Inject()(implicit frontendAppConfig: FrontendAppConfig
...
     case Some(_) => regulatorDocsRoutes.IsApprovedGoverningDocumentController.onPageLoad(mode)
     case _ => routes.SessionExpiredController.onPageLoad()
   }
 
   private def isApprovedGoverningDocumentPageNav(userAnswers: UserAnswers, mode: Mode): Call = userAnswers.get(IsApprovedGoverningDocumentPage) match {
+    case Some(true) if mode == CheckMode => userAnswers.get(HasCharityChangedPartsOfGoverningDocumentPage) match {
+      case Some(_) if userAnswers.get(HasCharityChangedPartsOfGoverningDocumentPage).isDefined =>
+        regulatorDocsRoutes.GoverningDocumentSummaryController.onPageLoad()
+      case _ => regulatorDocsRoutes.HasCharityChangedPartsOfGoverningDocumentController.onPageLoad(mode)
+    }
```
Comment (RafSobol) [inline app/navigation/DocumentsNavigator.scala:103]: do we need this pattern match here? wouldn't 
`      if (userAnswers.get(HasCharityChangedPartsOfGoverningDocumentPage).isDefined) {
        regulatorDocsRoutes.GoverningDocumentSummaryController.onPageLoad()
      } else {
        regulatorDocsRoutes.HasCharityChangedPartsOfGoverningDocumentController.onPageLoad(mode)
      }
` do the same thing?

File: app/navigation/DocumentsNavigator.scala [hunk 2]
```diff
@@ -85,26 +89,34 @@ class DocumentsNavigator @Inject()(implicit frontendAppConfig: FrontendAppConfig
...
   }
 
-  private def hasCharityChangedPartsOfGoverningDocumentPageNav(userAnswers: UserAnswers, mode: Mode): Call = userAnswers.get(HasCharityChangedPartsOfGoverningDocumentPage) match {
-    case Some(true) => regulatorDocsRoutes.SectionsChangedGoverningDocumentController.onPageLoad(mode)
+  private def hasCharityChangedPartsOfGoverningDocumentPageNav(userAnswers: UserAnswers, mode: Mode): Call =
+    userAnswers.get(HasCharityChangedPartsOfGoverningDocumentPage) match {
+    case Some(true) if mode == CheckMode && userAnswers.get(SectionsChangedGoverningDocumentPage).isDefined =>
+      regulatorDocsRoutes.GoverningDocumentSummaryController.onPageLoad()
+    case Some(true)  => regulatorDocsRoutes.SectionsChangedGoverningDocumentController.onPageLoad(mode)
     case Some(false) => regulatorDocsRoutes.GoverningDocumentSummaryController.onPageLoad()
```
Comment (RafSobol) [inline app/navigation/DocumentsNavigator.scala:114]: might be a little cleaner if `Some(true) if ...` and `Some(false)` are collapsed into case x | y? not too bothered about this one though

File: app/navigation/DocumentsNavigator.scala (hunk 1 above)
Comment (abhijeetardale) [reply app/navigation/DocumentsNavigator.scala:103]: fixed

File: app/navigation/DocumentsNavigator.scala (hunk 2 above)
Comment (abhijeetardale) [reply app/navigation/DocumentsNavigator.scala:114]: if is only applicable for some (true) hence not combined

**Review by RafSobol — APPROVED** [review]

**Pull request merged**